parcels/interaction/baseinteractionkernel.py

```diff
@@ -0,0 +1,145 @@
+import re
```

Comment:
The name of this file follows the convention of e.g. `basekernel`, `baseparticleset` as in the rest of parcels. If we want to change this for consistency, I would suggest to remove the underscores from the other filenames, rather than to add them here.